lib/protobuf/field/varint_field.rb

```diff
@@ -37,11 +37,17 @@ def self.encode(value)
       #
 
       def acceptable?(val)
-        (val > self.class.min || val < self.class.max)
+        int_val = coerce!(val)
+        int_val >= self.class.min && int_val <= self.class.max
       rescue
         false
       end
 
+      def coerce!(val)
+        return val.to_i if val.is_a?(Numeric)
```

Comment:
It is required to convert a float to an integer, unless that should be disallowed.

Also, `Integer('0x10')` produces 16. IMO, it should return an error and hence requires `Integer(val, 10)` which throws an error if `val` is of type Numeric.